src/httpd_jsonapi.c

```diff
@@ -2634,6 +2634,111 @@ jsonapi_reply_library_album_tracks(struct httpd_request *hreq)
   return HTTP_OK;
 }
 
+static int
+jsonapi_reply_library_tracks_get_byid(struct httpd_request *hreq)
+{
+  time_t db_update;
+  struct query_params query_params;
+  const char *track_id;
+  struct db_media_file_info dbmfi;
+  json_object *reply = NULL;
+  int ret = 0;
+
+  db_update = (time_t) db_admin_getint64(DB_ADMIN_DB_UPDATE);
+  if (db_update && httpd_request_not_modified_since(hreq->req, &db_update))
+    return HTTP_NOTMODIFIED;
+
+
+  track_id = hreq->uri_parsed->path_parts[3];
+
+  memset(&query_params, 0, sizeof(struct query_params));
+
+  query_params.type = Q_ITEMS;
+  query_params.sort = S_ALBUM;
+  query_params.filter = db_mprintf("(f.id = %q)", track_id);
```

Comment:
A bit unclean ... if track_to_json() could take mfi you could use db_file_fetch_byid() which would be nicer + shorter

Reply:
I thought about using `db_file_fetch_byid` but did not want to duplicate the `track_to_json` function just for this one use case. Using `db_file_fetch_byid` would make this function shorter and easier to read, it would also require a new function to map the data to json. And if the track object gets a new attribute in the future, two functions would need to be updated.

But i don't have a strong opinion on that and will change it, if you think, using `db_file_fetch_byid` is the better approach.

Reply:
It's up to you, I think the current way is ok.

We might consider exposing mfi_cols_map (and siblings), then you wouldn't need to keep track_to_json in sync with the track object. Maybe just expose them via some iterator function to avoid moving the whole thing to db.h. What do you think of that